lua/websocket/server.lua

```diff
@@ -166,15 +245,46 @@ function SERVER:Think()
 		return
 	end
 
+	for i = 1, self.numconnections do
+		local connection = self.connections[i]
+		while #connection.sendBuffer > 0 and IsValid(connection) do
+			local toSend = connection.sendBuffer[1]
+			local sentLen, err = connection.socket:send(toSend)
+
+			if err then
```

Comment:
You should check if the first return is nil instead. That's what tells you there's an error condition.